src/select/select.js

```diff
@@ -289,16 +300,19 @@ class Select extends React.Component<PropsT, StatelessStateT> {
           />
         )}
         {selectedOptions.map(option => (
-          <Tag key={option.id} $multiple={multiple}>
+          <Tag key={option.id} $multiple={multiple} disabled={disabled}>
             {this.getSelectedOptionLabel(option)}
             {multiple && (
               <SearchIcon
+                disabled={disabled}
                 onClick={e => {
-                  this.setState({
-                    selectedOptions: this.state.selectedOptions.filter(
-                      selectedOption => selectedOption.id !== option.id,
-                    ),
-                  });
+                  if (!disabled) {
```

Comment:
there is a better way to prevent events from happening. You capture event on top level DOM element and prevent it from going down 
![image](https://user-images.githubusercontent.com/1851469/45175179-95831380-b1c1-11e8-898c-62c6f1f1e0c6.png)